### PR TITLE
Fix dataset checks

### PR DESCRIPTION
**Pull request by daniel-thom**
Fixes two small bugs in the query and derived-dataset code.

### CODE REVIEW

File: dsgrid/utils/dataset.py [hunk 1]
```diff
@@ -165,7 +165,9 @@ def check_null_value_in_unique_dimension_rows(dim_table, exclude_columns=None):
 def is_noop_mapping(records: pyspark.sql.DataFrame) -> bool:
     """Return True if the mapping is a no-op."""
     return records.filter(
-        (records.from_id != records.to_id) | (records.from_fraction != 1.0)
+        (records.to_id.isNull() & ~records.from_id.isNull())
```
Comment (elainethale) [inline dsgrid/utils/dataset.py:168]: Should there also be:

```
(~records.to_id.isNull() & records.from_id.isNull())
```

?

**Review by elainethale — APPROVED** [review]
Minor question. Otherwise looks good to me.

File: dsgrid/utils/dataset.py (hunk 1 above)
Comment (elainethale) [reply dsgrid/utils/dataset.py:168]: Good point. Null on the from_id side probably doesn't happen? That would correspond to being explicit about "this thing exists in the project but we aren't mapping anything to it."

Comment (daniel-thom) [reply dsgrid/utils/dataset.py:168]: Sounds good. It would be annoying bug to figure out and the extra check won't harm anything.

**Pull request merged**